### PR TITLE
Fix wrong links

### PR DESCRIPTION
**Pull request by mathbunnyru**
I used sphinx link check (`sphinx-build -b linkcheck docs/ docs/_build/` or `make linkcheck-docs` in our Makefile) to get rid of broken links and redirects.

### CODE REVIEW

File: docs/contributing/stacks.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Following these steps will:
 1. Setup a project on GitHub containing a Dockerfile based on any of the images we provide.
 2. Configure GitHub Actions to build and test your image when users submit pull requests to your repository.
 3. Configure Docker Hub to build and host your images for others to use.
-4. Update the [list of community stacks](../using/selecting.html#community-stacks) in this documentation to include your image.
+4. Update the [list of community stacks](../using/selecting.md#community-stacks) in this documentation to include your image.
```
Comment (mathbunnyru) [inline docs/contributing/stacks.md:12]: This type of links works well with code editors and with Github.

**Comment by trallard** [comment]
LGMT - could this be added as a pre-commit hook? or as part of the `Build Sphinx Documentation` workflow?

**Comment by consideRatio** [comment]
In other projects we have a linkcheck as part of the github workflow that triggers on changes to docs.

See https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/main/.github/workflows/test-docs.yaml#L25-L43

Note the command `make linkcheck SPHINXOPTS='--color -W --keep-going'` --- force colors, error on warnings, don't stop on first error.

**Comment by mathbunnyru** [comment]
I see, I will later implement this as part of the CI in a separate PR, thanks!

File: docs/conf.py [hunk 2]
```diff
@@ -85,3 +85,5 @@
     ".md": "markdown",
 }
 pygments_style = "sphinx"
+
+myst_heading_anchors = 3
```
Comment (consideRatio) [inline docs/conf.py:89]: What does this do?

> Enable auto-generated heading anchors, up to a maximum level, [see here](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#syntax-header-anchors) for details.

Ah...

```suggestion
# MyST configuration reference: https://myst-parser.readthedocs.io/en/latest/sphinx/reference.html
myst_heading_anchors = 3
```

**Review by consideRatio — APPROVED** [review]
LGTM!

**Pull request merged**